### ISSUE TITLE
hack/test-end-to-end.sh doesn't kill openshift process

### ISSUE THREAD

**Issue by deads2k** [openshift/origin#844]
I get messages like:
```
[INFO] Tearing down test
kill: cannot find process "- 4880"
[INFO] Stopping k8s docker containers
```

and then after the test is over, I see:
```
[deads@deads-dev-01 origin]$ ps ax | grep opens
 4900 pts/1    S      0:00 sudo env PATH=hack/../_output/local/go/bin:/sbin:/bin:/usr/sbin:/usr/bin OPENSHIFT_ON_PANIC=crash openshift start --listen=https://0.0.0.0:8443 --public-master=https://localhost --hostname=127.0.0.1 --volume-dir=/tmp/openshift-e2e/volumes --etcd-dir=/tmp/openshift-e2e/etcd --cert-dir=/tmp/openshift-e2e/certs --loglevel=4
 4905 pts/1    Sl     0:24 openshift start --listen=https://0.0.0.0:8443 --public-master=https://localhost --hostname=127.0.0.1 --volume-dir=/tmp/openshift-e2e/volumes --etcd-dir=/tmp/openshift-e2e/etcd --cert-dir=/tmp/openshift-e2e/certs --loglevel=4
10400 ?        Ssl    0:00 /usr/bin/openshift-router --template=/var/lib/haproxy/conf/haproxy-config.template --reload=/var/lib/haproxy/reload-haproxy --loglevel=4
10492 ?        Ssl    0:00 /usr/bin/openshift-sti-build
```

@smarterclayton Is this a personal problem?  I see that the kill changed in: https://github.com/openshift/origin/pull/822, but I don't want to just revert it if it's supposed to work.

**Comment by deads2k** [comment]
Well, here anyway: https://github.com/openshift/origin/commit/2fbc30d2fb3778ffb62ba56ebfbd1169dd4498c0#diff-11587e7dcf72b238b397919567c4707e.  I seem to have followed the chain to the wrong pull somehow.

**Comment by smarterclayton** [comment]
Cannot find process, eh?  Can you add `sudo ps -fo "%p %r %c %a"` right before and after the kill in your local copy and see if you can recreate?

**Comment by smarterclayton** [comment]
This could be the "kill kills itself" issue people alluded to.

**Comment by deads2k** [comment]
```
[INFO] Tearing down test
[INFO] BeforeKill
  PID  PGID COMMAND         COMMAND
21569 21569 sudo            sudo baobab
21573 21569  \_ baobab       \_ baobab
 8559  8559 sudo            sudo hack/test-end-to-end.sh
 8566  8559  \_ test-end-to  \_ /bin/bash hack/test-end-to-end.sh
 8585  8559      \_ sudo         \_ sudo env PATH=hack/../_output/local/go/bin:/sbin:/bin:/usr/sbin:/usr/bin OPENSHIFT_ON_PANIC=crash openshift
 8589  8559      |   \_ ope      |   \_ openshift start --listen=https://0.0.0.0:8443 --public-master=https://localhost --hostname=127.0.0.1 --
17350  8559      \_ sudo         \_ sudo ps -fo %p %r %c %a
17353  8559          \_ ps           \_ ps -fo %p %r %c %a
17842 17842 sudo            sudo hack/test-end-to-end.sh
 1226  1226 Xorg            /usr/bin/Xorg :0 -background none -verbose -auth /run/gdm/auth-for-gdm-jdXYLj/database -seat seat0 -nolisten tcp vt
21257 21252 dbus-launch     dbus-launch --autolaunch=69165d99e47640bf887f9cbf1160135e --binary-syntax --close-stderr
kill: cannot find process "- 8559"
[INFO] AfterKill
  PID  PGID COMMAND         COMMAND
21569 21569 sudo            sudo baobab
21573 21569  \_ baobab       \_ baobab
 8559  8559 sudo            sudo hack/test-end-to-end.sh
 8566  8559  \_ test-end-to  \_ /bin/bash hack/test-end-to-end.sh
 8585  8559      \_ sudo         \_ sudo env PATH=hack/../_output/local/go/bin:/sbin:/bin:/usr/sbin:/usr/bin OPENSHIFT_ON_PANIC=crash openshift
 8589  8559      |   \_ ope      |   \_ openshift start --listen=https://0.0.0.0:8443 --public-master=https://localhost --hostname=127.0.0.1 --
17363  8559      \_ sudo         \_ sudo ps -fo %p %r %c %a
17365  8559          \_ ps           \_ ps -fo %p %r %c %a
17842 17842 sudo            sudo hack/test-end-to-end.sh
 1226  1226 Xorg            /usr/bin/Xorg :0 -background none -verbose -auth /run/gdm/auth-for-gdm-jdXYLj/database -seat seat0 -nolisten tcp vt
21257 21252 dbus-launch     dbus-launch --autolaunch=69165d99e47640bf887f9cbf1160135e --binary-syntax --close-stderr
[INFO] Stopping k8s docker containers
```

**Comment by smarterclayton** [comment]
I have a fix

**Comment by smarterclayton** [comment]
Fixed by #863 probably

**Comment by smarterclayton** [comment]
Now is actually fixed.
